Upstream patch

"Fixing bug #1494773 (compiles only on i368/amd64 machines) by providing a dummy implementation for other archs "
See http://sourceforge.net/tracker/index.php?func=detail&aid=1494773&group_id=1310&atid=101310

===================================================================
--- ./udelay.h (revision 600)
+++ ./udelay.h (revision 712)
@@ -1,3 +1,3 @@
-/* $Id: lcd4linux-0.10.0-no-x86-asm.patch,v 1.1 2007/02/18 23:02:56 rbu Exp $
+/* $Id: lcd4linux-0.10.0-no-x86-asm.patch,v 1.1 2007/02/18 23:02:56 rbu Exp $
  *
  * short delays 
@@ -24,4 +24,7 @@
  *
  * $Log: lcd4linux-0.10.0-no-x86-asm.patch,v $
  * Revision 1.1  2007/02/18 23:02:56  rbu
  * Introducing LCD_DEVICES, move of config file, code should compile on non-x86 again, new init-script and other fixes
  * (Portage version: 2.1.2-r9)
  *
+ * Revision 1.12  2006/09/13 20:07:59  entropy
+ * Fixing bug #1494773 (compiles only on i368/amd64 machines) by providing a dummy implementation for other archs
+ *
  * Revision 1.11  2005/12/12 09:08:08  reinelt
  * finally removed old udelay code path; read timing values from config
@@ -80,5 +83,12 @@
 static inline void rep_nop(void)
 {
+# if defined(__i386) || defined(__i386__) || defined(__AMD64__) || defined(__x86_64__) || defined(__amd64__)
+    /* intel or amd64 arch, the "rep" and "nop" opcodes are available */
     __asm__ __volatile__("rep; nop");
+# else
+    /* other Arch, maybe add core cooldown code here, too. */
+    do {
+    } while (0);
+# endif
 }
 
